client/network/src/config.rs

```diff
@@ -139,6 +136,9 @@ where
 
 	/// Optional warp sync protocol config.
 	pub warp_sync_protocol_config: Option<RequestResponseConfig>,
+
+	/// Request response protocol configurations
+	pub request_response_protocol_configs: Vec<RequestResponseConfig>,
```

Comment:
All those other `RequestResponseConfig` should be removable or? And instead being added here? However this can be done in a later pr.

Reply:
Yes they should ideally all be here but unfortunately it will be big change to pass them all using this so I'd rather do it in a separate PR.